src/hooks/useMenuKeyboardNavigation.ts

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import {iterateFocusableElements} from '@primer/behaviors/utils'
+import {useMenuInitialFocus} from './useMenuInitialFocus'
+import {useTypeaheadFocus} from './useTypeaheadFocus'
+import {MenuContextProps} from '../ActionMenu'
+
+/**
+ * Keyboard navigation is a mix of 4 hooks
+ * 1. useMenuInitialFocus
+ * 2. useTypeaheadFocus
+ * 3. useCloseMenuOnTab
+ * 4. useMoveFocusToMenuItem
+ */
+export const useMenuKeyboardNavigation = (
+  open: boolean,
+  onOpen: MenuContextProps['onOpen'],
+  onClose: MenuContextProps['onClose'],
+  containerRef: React.RefObject<HTMLElement>,
+  anchorRef: React.RefObject<HTMLElement>
+) => {
+  const {openWithFocus} = useMenuInitialFocus(open, onOpen, containerRef, anchorRef)
+  useTypeaheadFocus(open, containerRef, anchorRef)
+  useCloseMenuOnTab(open, onClose, containerRef, anchorRef)
+  useMoveFocusToMenuItem(open, containerRef, anchorRef)
+
+  return {openWithFocus}
+}
+
+/**
+ * When Tab or Shift+Tab is pressed, the menu should close
+ * and the focus should naturally move to the next item
+ */
+const useCloseMenuOnTab = (
+  open: boolean,
+  onClose: MenuContextProps['onClose'],
+  containerRef: React.RefObject<HTMLElement>,
+  anchorRef: React.RefObject<HTMLElement>
+) => {
+  React.useEffect(() => {
+    const container = containerRef.current
+    const anchor = anchorRef.current
+
+    const handler = (event: KeyboardEvent) => {
+      if (open && event.key === 'Tab') onClose?.('tab')
+    }
+
+    container?.addEventListener('keydown', handler)
+    anchor?.addEventListener('keydown', handler)
+    return () => {
+      container?.removeEventListener('keydown', handler)
+      anchor?.removeEventListener('keydown', handler)
+    }
+  }, [open, onClose, containerRef, anchorRef])
+}
+
+/**
+ * When Arrow Keys are pressed and the focus is on the anchor,
+ * focus should move to a menu item
+ */
+const useMoveFocusToMenuItem = (
```

Comment:
comment: When menu is open with keyboard, the focus moves to the menu item based on the key pressed. However, when menu is opened with click, the focus stays on the anchor. 

Pressing ArrowDown or ArrowUp should move the focus to the right menu item

https://user-images.githubusercontent.com/1863771/159458645-806e05cc-c264-45e4-b9eb-c4828cb8a7f7.mov